Scroller/tests/Scroller-specs.js

```diff
@@ -19,167 +21,159 @@ describe('Scroller', () => {
+			expect(previousScrollbarButton).toHaveClass('horizontal');
+		});
 
-		test(
-			'should set "aria-label" to next scroll button in the vertical scroll bar',
-			() => {
-				const label = 'custom button aria label';
-				const subject = mount(
-					<Scroller
-						horizontalScrollbar="visible"
-						verticalScrollbar="visible"
-						scrollDownAriaLabel={label}
-					>
-						{contents}
-					</Scroller>
-				);
-
-				const expected = label;
-				const actual = subject.find('ScrollButton').at(1).prop('aria-label');
-
-				expect(actual).toBe(expected);
-			}
-		);
-	});
+		test('should set "aria-label" to next scroll button in the horizontal scroll bar', () => {
+			const label = 'custom button aria label';
+			render(
+				<Scroller
+					horizontalScrollbar="visible"
+					scrollRightAriaLabel={label}
+					verticalScrollbar="visible"
+				>
+					{contents}
+				</Scroller>
+			);
 
-	describe('ScrollerBase API', () => {
-		test('should call onUpdate when Scroller updates', () => {
-			const handleUpdate = jest.fn();
-			const subject = shallow(
-				<ScrollerBase
-					onUpdate={handleUpdate}
+			const nextScrollbarButtonLabel = screen.getByLabelText(label);
+			const nextScrollbarButton = screen.getByLabelText(label).parentElement;
+
+			expect(nextScrollbarButtonLabel).toBeInTheDocument();
+			expect(nextScrollbarButton).toHaveClass('horizontal');
+		});
+
+		test('should set "aria-label" to previous scroll button in the vertical scroll bar', () => {
+			const label = 'custom button aria label';
+			render(
+				<Scroller
+					horizontalScrollbar="visible"
+					scrollUpAriaLabel={label}
+					verticalScrollbar="visible"
+				>
+					{contents}
+				</Scroller>
+			);
+
+			const previousScrollbarButtonLabel = screen.getByLabelText(label);
+			const previousScrollbarButton = screen.getByLabelText(label).parentElement;
+
+			expect(previousScrollbarButtonLabel).toBeInTheDocument();
+			expect(previousScrollbarButton).toHaveClass('vertical');
+		});
+
+		test('should set "aria-label" to next scroll button in the vertical scroll bar', () => {
+			const label = 'custom button aria label';
+			render(
+				<Scroller
+					horizontalScrollbar="visible"
+					verticalScrollbar="visible"
+					scrollDownAriaLabel={label}
 				>
 					{contents}
-				</ScrollerBase>
+				</Scroller>
 			);
 
-			subject.setProps({children: ''});
+			const nextScrollbarButtonLabel = screen.getByLabelText(label);
+			const nextScrollbarButton = screen.getByLabelText(label).parentElement;
 
-			const expected = 1;
-			const actual = handleUpdate.mock.calls.length;
+			expect(nextScrollbarButtonLabel).toBeInTheDocument();
+			expect(nextScrollbarButton).toHaveClass('vertical');
+		});
+	});
+
+	// Left this as skipped because we can't check at this moment the onUpdate call with react-testing-library
+	describe('ScrollerBase API', () => {
+		test.skip('should call onUpdate when Scroller updates', async () => {
```

Comment:
This test should render `ScrollerBase` instead of `Scroller` since `Scrollable` will override `onUpdate` function. So you need to import `ScrollerBase` above.
Also, when we rerender the component, you need to pass `onUpdate` prop as well.
So it should look like,
```
test('should call onUpdate when Scroller updates', async () => {
	const handleUpdate = jest.fn();
	const {rerender} = render(<ScrollerBase onUpdate={handleUpdate}>{contents}</ScrollerBase>);

	rerender(
		<ScrollerBase onUpdate={handleUpdate}>{''}</ScrollerBase>
	);

	await waitFor(() => expect(handleUpdate).toHaveBeenCalled());
});
```
But it will fail to render since we need a guard in `initUiRef` function in `ScrollerBase` at line 349.
So please add a guard like below.
```
initUiRef = (ref) => {
	if (ref) {
		this.uiRefCurrent = ref;
		if (this.props.initUiChildRef) {
			this.props.initUiChildRef(ref);
		}
	}
};
```
Hope this helps.

Reply:
Your solution did the trick. Thank you for your help!